kernel/device.c

```diff
@@ -64,9 +64,9 @@ void z_sys_init_run_level(int32_t level)
 			z_object_init(dev);
 		}
 
-		if ((entry->init(dev) == 0) && (dev != NULL)) {
-			/* Initialization was successful.
-			 * Set the init status bit so device is declared ready.
+		if ((entry->init(dev) != 0) && (dev != NULL)) {
```

Comment:
Shouldn't the NULL check for the dev pointer be negated as well?

Reply:
@henrikbrixandersen: I think that is fine as is, the check is there because `dev` is used to calculate offset in `__device_init_status_start`.